File: leetcode-submissions/2149_Rearrange_Array_Elements_by_Sign.ts
function rearrangeArray(nums: number[]): number[] {
    const negative:number[]=[];
    const positive:number[]=[];
    for(const ele of nums){
        if(ele<0){
            negative.push(ele);
        }else{
            positive.push(ele);

        }
    }
    let i:number=0;
    const ans:number[]=[];

    while(i<positive.length){
        ans.push(positive[i]);
        ans.push(negative[i]);
        i++;
    }
    return ans;
};

File: leetcode-submissions/2149_Rearrange_Array_Elements_by_Sign.cpp
class Solution {
public:
    vector<int> rearrangeArray(vector<int>& nums) {
        vector<int>n1;
        vector<int>n2;
        for(int i=0;i<nums.size();i++){
            if(nums[i]<0){
                n2.push_back(nums[i]);
            }else{
                n1.push_back(nums[i]);
            }
        }
        vector<int>ans;
        for(int i=0;i<n1.size();i++){
            ans.push_back(n1[i]);
            ans.push_back(n2[i]);


        }
        return ans;
        
    }
};